crates/sui-types/src/messages.rs

```diff
@@ -1871,40 +1842,95 @@ pub struct EpochEnvelop<S> {
     pub auth_sign_info: S,
 }
 
+pub type GenesisEpoch = EpochEnvelop<EmptySignInfo>;
 pub type SignedEpoch = EpochEnvelop<AuthoritySignInfo>;
 pub type CertifiedEpoch = EpochEnvelop<AuthorityStrongQuorumSignInfo>;
 
+impl GenesisEpoch {
+    pub fn new(committee: Committee) -> Self {
+        Self {
+            epoch_info: EpochInfo::new(committee, 0),
+            auth_sign_info: EmptySignInfo {},
+        }
+    }
+}
+
 impl SignedEpoch {
     pub fn new(
-        epoch: EpochId,
-        next_epoch_committee: Committee,
+        committee: Committee,
         authority: AuthorityName,
         secret: &dyn signature::Signer<AuthoritySignature>,
-        last_checkpoint: CheckpointSequenceNumber,
-    ) -> SuiResult<Self> {
-        let epoch_info = EpochInfo::new(epoch, next_epoch_committee, last_checkpoint)?;
+        first_checkpoint: CheckpointSequenceNumber,
+    ) -> Self {
+        let epoch = committee.epoch;
+        let epoch_info = EpochInfo::new(committee, first_checkpoint);
         let signature = AuthoritySignature::new(&epoch_info, secret);
-        Ok(Self {
+        Self {
             epoch_info,
             auth_sign_info: AuthoritySignInfo {
-                epoch,
+                // A epoch is always authenticated by validators from the previous epoch.
+                // This is critical: we won't be able to use the same committee to verify itself,
+                // hence we rely on the committee from the previous epoch to verify the current
+                // epoch data structure.
+                epoch: epoch - 1,
                 authority,
                 signature,
             },
-        })
+        }
     }
 
     pub fn verify(&self, committee: &Committee) -> SuiResult {
```

Comment:
What committee is the one given in the arguments. Since we now are handling committees from diff epochs we may want to call them specific names such as "prev_epoch_committee" if this is what it is.